plugins/notifiers/slack/client_test.go

```diff
@@ -0,0 +1,67 @@
+package slack
+
+import (
+	"bytes"
+	"errors"
+	"github.com/odpf/guardian/domain"
+	"github.com/odpf/guardian/mocks"
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/suite"
+	"io/ioutil"
+	"net/http"
+	"testing"
+)
+
+type ClientTestSuite struct {
+	suite.Suite
+	mockHttpClient *mocks.HTTPClient
+	accessToken    string
+	messages       domain.NotificationMessages
+	slackIDCache   map[string]string
+	notifier       notifier
+}
+
+func (s *ClientTestSuite) setup() {
+	s.mockHttpClient = new(mocks.HTTPClient)
+	s.accessToken = "XXXXX-TOKEN-XXXXX"
+	s.messages = domain.NotificationMessages{
+		AppealRejected: "[{\"type\":\"section\",\"text\":{\"type\":\"mrkdwn\",\"text\":\"Your appeal to {{.resource_name}} with role {{.role}} has been rejected\"}}]",
+	}
+	s.slackIDCache = map[string]string{}
+	s.notifier = notifier{
+		accessToken:  s.accessToken,
+		slackIDCache: s.slackIDCache,
+		Messages:     s.messages,
+		httpClient:   s.mockHttpClient,
+	}
+}
+
+func TestClient(t *testing.T) {
+	suite.Run(t, new(ClientTestSuite))
+}
+
+func (s *ClientTestSuite) TestNotify() {
```

Comment:
can we add positive case to cover reading the template file to work as expected?

Reply:
Done